### PR TITLE
feat(TDC-4779): add input text for the quick faceted search

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

add input text for the quick faceted search

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: src/components/BasicSearch/BasicSearch.component.js [hunk 1]
```diff
@@ -76,6 +82,25 @@ const BasicSearch = ({
...
+					dispatch(
+						BADGES_ACTIONS.addWithValue(
+							generateBadge(operators)(facet),
+							operators.find(({ name }) => name === DEFAULT_QUICKSEARCH_OPERATOR) ||
+								operators[0],
+							value,
+						),
+					);
+				}}
+			/>
```
Comment (romainseb) [inline src/components/BasicSearch/BasicSearch.component.js:103]:
```suggestion
			{quicksearchable.length > 0 && <QuickSearchInput
				t={t}
				className={css('tc-basic-search-quicksearch')}
				facets={quicksearchable}
				onSelect={(facet, value) => {
					const operators = getOperatorsFromDict(
						operatorsDictionary,
						get(facet, 'metadata.operators'),
					);
					dispatch(
						BADGES_ACTIONS.addWithValue(
							generateBadge(operators)(facet),
							operators.find(({ name }) => name === DEFAULT_QUICKSEARCH_OPERATOR) ||
								operators[0],
							value,
						),
					);
				}}
			/>}
```

**Comment by olliew-talend** [comment]
Some questions about the UI behaviour:

1. When I open the faceted search panel the quicksearch input and badge seem to 'vibrate' a little after they appear. I thought it was my eyes being funny but no.
2. When you open the faceted search panel, the search icon retains the focus style which makes it look a little strange. This doesn't happen for the display mode icons
3. The 'Search in' dropdown header doesn't seem to match the Frontify docs. The font colour, margins and border seem different
4. If I click on the 'search in' header, the input loses focus, my entry is deleted and the dropdown closes. This should probably not happen if my click is within the context of the quicksearch
5. Accessibility: After entering something in the quick search input and having the dropdown appear, I can't access the dropdown with the arrow keys on the keyboard and pressing tab moves focus to the 'add filter' badge which deletes my quicksearch input

Also: An existing issue but I still find it super strange that when you delete your last active facet, the whole faceted search bar closes. Surely we should stop this as we can't know for certain that the user doesn't want to add a new facet?

**Comment by olliew-talend** [comment]
The quicksearch (and existing faceted search) does not return results for datasets with special chars in their name. See sceenshots below. This should probably be raised as a separate bug.

Screenshot 2020-11-25 at 09 44 15
Screenshot 2020-11-25 at 09 47 52

**Review by romainseb — APPROVED** [review]

**Comment by ncomont** [comment]
> When I open the faceted search panel the quicksearch input and badge seem to 'vibrate' a little after they appear. I thought it was my eyes being funny but no.

This is a known issue

> When you open the faceted search panel, the search icon retains the focus style which makes it look a little strange. This doesn't happen for the display mode icons

Same

> The 'Search in' dropdown header doesn't seem to match the Frontify docs. The font colour, margins and border seem different

This is a known TUI issue : the Typeahead has to be reworked. Chloé and Maxime are aware of that. We can't fix it without doing faceted-specific code.

> If I click on the 'search in' header, the input loses focus, my entry is deleted and the dropdown closes. This should probably not happen if my click is within the context of the quicksearch

This is expected. According to Maxime : "si on est d’accord sur le fait que l’input se vide à la sélection d’une facet ça me va"

> If I enter text in the input and then click away my text is removed from the input. It would maybe be better if the dropdown was removed but the input text remained. If the input is re-focussed, the dropdown could be called back with existing text

It seems to be ok for the UX, but some adjustments will be done after the demo, I guess.

> Accessibility: After entering something in the quick search input and having the dropdown appear, I can't access the dropdown with the arrow keys on the keyboard and pressing tab moves focus to the 'add filter' badge which deletes my quicksearch input

AFAIK this is not expected, but I agree, it could be great. This should be a Typeahead behaviour we'll have to improve during the Typeahead rework.

**Comment by ncomont** [comment]
> The quicksearch (and existing faceted search) does not return results for datasets with special chars in their name. See sceenshots below. This should probably be raised as a separate bug.

I can't reproduce on my side. Are you sure this is the same char in the title and in your search ? Anyway it wouldn't be related to this PR.

accent

**Comment by olliew-talend** [comment]
I checked again to be sure it was the same character and it seems that it's only a problem with datasets that are created as a direct drag and drop file upload, not those created via the 'add dataset' button. This is certainly a separate bug and maybe related to my current encoding issue. I will raise separately. 

Okay for my other questions, I might raise some with Maxime for further dicussion.

**Review by olliew-talend — APPROVED** [review]

**Pull request merged**